pkg/ctl/cmdutils/configfile.go

```diff
@@ -761,3 +763,84 @@ func NewDeleteIAMServiceAccountLoader(cmd *Cmd, sa *api.ClusterIAMServiceAccount
 
 	return l
 }
+
+// NewUpdateNodegroupLoader will load config or use flags for 'eksctl update nodegroup'.
+func NewUpdateNodegroupLoader(cmd *Cmd) ClusterConfigLoader {
+	l := newCommonClusterConfigLoader(cmd)
+
+	l.validateWithConfigFile = func() error {
+		var clusterConfig *api.ClusterConfig
+		var err error
+		if clusterConfig, err = eks.LoadConfigFromFile(l.ClusterConfigFile); err != nil {
+			return err
+		}
+
+		l := len(clusterConfig.ManagedNodeGroups)
+		if l < 1 {
+			return ErrMustBeSet("managedNodeGroups field")
+		}
+
+		var unsupportedFields []string
+		ng := clusterConfig.ManagedNodeGroups[0]
```

Comment:
do we not need to validate all managed nodegroups, since they are all getting updated?

Reply:
for now it's just 1 nodegroup but I'll open another PR right afterwards to iterate over all nodegroups! https://github.com/weaveworks/eksctl/pull/3828#discussion_r651658891